### PR TITLE
native build on aarch64

### PR DESCRIPTION
**Pull request by hadfl**
With these changes plus #76 and #77 I could successfully `dmake setup` on aarch64. I made sure, that cross building on x86 for aarch64 still works with this in place.

### CODE REVIEW

File: usr/src/tools/svc/libscf/Makefile [hunk 1]
```diff
@@ -59,7 +59,9 @@ LIBUUTIL =	$(SRC)/lib/libuutil
 CPPFLAGS +=	-DNATIVE_BUILD $(DTEXTDOM) \
 		-I$(SRC)/lib/libscf/inc -I$(COMDIR) -I$(LIBUUTIL)/common
 LDLIBS +=	-R '$$ORIGIN/../../lib/$(NATIVE_MACH)' -L$(ROOTONBLDLIBMACH) \
-		-luutil -lc -lgen -lnvpair -lsmbios
+		-luutil -lc -lgen -lnvpair
+$(NOT_AARCH64_BLD)LDLIBS += -lsmbios
+$(AARCH64_BLD)$(CROSS_BLD)LDLIBS += -lsmbios
```
Comment (richlowe) [inline usr/src/tools/svc/libscf/Makefile:64]: Why is this only needed in a cross build?

Comment (richlowe) [reply usr/src/tools/svc/libscf/Makefile:64]: How do you feel about

```diff
diff --git a/usr/src/tools/svc/libscf/Makefile b/usr/src/tools/svc/libscf/Makefile
index 8e1f3f84d3..a4605b271a 100644
--- a/usr/src/tools/svc/libscf/Makefile
+++ b/usr/src/tools/svc/libscf/Makefile
@@ -60,8 +60,10 @@ CPPFLAGS +=  -DNATIVE_BUILD $(DTEXTDOM) \
                -I$(SRC)/lib/libscf/inc -I$(COMDIR) -I$(LIBUUTIL)/common
 LDLIBS +=      -R '$$ORIGIN/../../lib/$(NATIVE_MACH)' -L$(ROOTONBLDLIBMACH) \
                -luutil -lc -lgen -lnvpair
-$(NOT_AARCH64_BLD)LDLIBS += -lsmbios
-$(AARCH64_BLD)$(CROSS_BLD)LDLIBS += -lsmbios
+
+i386_LDLIBS += -lsmbios
+LDLIBS += $($(NATIVE_MACH)_LDLIBS)
+
 ADJUNCT_LIBS +=        libsmbios.so libnvpair.so libgen.so libc.so

 .KEEP_STATE:
 ```

instead?

Comment (hadfl) [reply usr/src/tools/svc/libscf/Makefile:64]: This still works on a native build on `aarch64` and seems much cleaner.

**Comment by richlowe** [comment]
merged

**Pull request closed without merging**